### PR TITLE
Redo rspec

### PR DESCRIPTION
**Pull request by dasileker**
### Testing Ruby with RSpec - Enumerable methods
In this milestone, I used RSpec to create a testing file for our enumerable methods project.

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Amine Zerradi** 
Nice job so far with your first submission :muscle:  :muscle: 
Your project meets most of the requirements. You have done a great effort testing different scenarios :clap: :clap: :clap:  
We still need to fix some details:

- [x] For each method, test positive and negative scenarios, you might use "to_not_eql" or "to_not be_an".
- [x] the description of each test should be unique. The description of the test must be more explicit.
- [x] Use 'require_relative' instead of 'require'.
- [x] In your README file, explain how to run your tests.


Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

File: spec/enum_redo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  let(:nil_array) { [1, 2, nil, 3, 4, 5] }
+  let(:false_array) { [nil, nil, false] }
+  let(:hash) { { a: 1, b: 2, c: 3, d: 4, e: 5 } }
+  let(:arr) { [2, 4, 5] }
+  let(:block) { proc { |num| num } }
+  let(:my_proc) { proc { |num| num > 10 } }
+  let(:range) { (5..10) }
+
+  describe '#my_each' do
+    describe 'it should return to Enumerator if there no block' do
```
Comment (expjazz) [inline spec/enum_redo_spec.rb]: - [x] Each method should be wrapped in one describe block (sometimes you have more than one describe per method).

image

File: spec/enum_redo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+      it 'should not return to Enumerator if there no block' do
+        expect(range.my_each).to be_an Enumerator
+      end
+      it 'should not return to Enumerator if there no block' do
+        expect(hash.my_each).to be_an Enumerator
+      end
+      it 'should not return to Enumerator if there no block' do
+        expect(nil_array.my_each).to be_an Enumerator
+      end
+      it 'should not return to Enumerator if there no block' do
```
Comment (expjazz) [inline spec/enum_redo_spec.rb]: - [x] I can see that you are testing different cases, but your _"it"_ description remains the same. Kindly update that.

**Review by expjazz — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

Hey team :+1: 

You did a great job. Your tests are descriptive, and they cover a lot of different cases. However, we do have a few pending issues that we still need to address: 

- [x] Each method should be wrapped in one describe block

- [x] You have some repetition in your _it_ descriptions

- [x] I can see that you added negative scenarios for the _my_each_, but I believe that the previous TSE asked to add the negative scenarios for all the methods. 
image

- [x] You also have the wrong Git flow (you should have a development branch, and push the changes to that branch)

Kindly check the inline comments below to see more.

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

File: spec/enum_redo_spec.rb (hunk 1 above)
Comment (dasileker) [reply spec/enum_redo_spec.rb]: thanks, man I was sick, I was struggling to finish.  I hope now is good

File: spec/enum_redo_spec.rb (hunk 2 above)
Comment (dasileker) [reply spec/enum_redo_spec.rb]: sorry for that I was thinking that I should do it but it turns not necessary, and sorry for the workflow too. like I said I was sick

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Well done :clap: :clap: :clap: @dasileker, thanks for making the required changes :hammer: 

now you can merge your project is approved.

## Happy Coding :tophat:

**Pull request merged**